libs/ledger-live-common/src/hw/ftsFetchImage.ts

```diff
@@ -0,0 +1,204 @@
+                  imageChunk.length - 2
+                );
+
+                if (chunkStatus !== 0x9000) {
+                  // TODO: map all proper errors
+                  return subscriber.error(
+                    new TransportError(
+                      "Unexpected device responce",
+                      chunkStatus.toString(16)
+                    )
+                  );
+                }
+
+                imageBuffer = Buffer.concat([
+                  imageBuffer,
+                  imageChunk.slice(0, imageChunk.length - 2),
+                ]);
+
+                currentOffset += chunkSize;
+              }
+
+              const hexImage = await parseFtsImageFormat(imageBuffer);
+
+              subscriber.next({ type: "imageFetched", hexImage });
+
+              subscriber.complete();
+            }),
+            catchError((e: unknown) => {
+              if (
+                e instanceof DeviceOnDashboardExpected ||
+                (e &&
+                  e instanceof TransportStatusError &&
+                  [0x6e00, 0x6d00, 0x6e01, 0x6d01, 0x6d02].includes(
+                    // @ts-expect-error typescript not checking agains the instanceof
+                    e.statusCode
+                  ))
+              ) {
+                return from(getAppAndVersion(transport)).pipe(
+                  concatMap((appAndVersion) => {
+                    return !isDashboardName(appAndVersion.name)
+                      ? attemptToQuitApp(transport, appAndVersion)
+                      : of<FetchImageEvent>({
+                          type: "appDetected",
+                        });
+                  })
+                );
+              }
+              return throwError(e);
+            })
+          )
+          .subscribe(subscriber);
+
+        return () => {
+          timeoutSub.unsubscribe();
+          sub.unsubscribe();
+        };
+      })
+  );
+
+  return sub as Observable<FetchImageEvent>;
+}
+
+// transforms from a FTS binary image format to an LLM hex string format
+const parseFtsImageFormat: (ftsImageBuffer: Buffer) => Promise<string> = async (
+  ftsImageBuffer
+) => {
+  // const width = ftsImageBuffer.readUint16LE(0); // always 400
+  // const height = ftsImageBuffer.readUint16LE(2); // always 672
+  const bppCompressionByte = ftsImageBuffer.readUInt8(4);
+
+  // const bpp = bppCompressionByte >> 4; // always 2
+  const compression = bppCompressionByte & 0x0f;
+
+  const dataLengthBuffer = Buffer.from([
+    ftsImageBuffer.readUInt8(5),
+    ftsImageBuffer.readUInt8(6),
+    ftsImageBuffer.readUInt8(7),
+    0x00,
+  ]);
+
+  const dataLength = dataLengthBuffer.readUInt32LE();
+  const imageData = ftsImageBuffer.slice(8);
+
+  if (compression === 0) {
+    return imageData.toString("hex");
+  }
+
+  let uncompressedImageData = Buffer.from([]);
+
+  let offset = 0;
+  while (offset < dataLength) {
+    const currentChunkSize = imageData.readUInt16LE(offset);
+    offset += 2;
```

Comment:
why `+2` here ? should we use constant value to explain/document some of the integer values ?